pkg/yaml/yaml.go

```diff
@@ -24,15 +26,62 @@ var (
 	cleanContains = []string{
 		"cattle.io/",
 	}
+	buffSize = 4096
 )
 
+// Unmarshal decodes YAML bytes into document (as defined by
+// the YAML spec) then converting it to JSON via
+// "k8s.io/apimachinery/pkg/util/yaml".YAMLToJSON and then decoding the json in the the v interface{}
 func Unmarshal(data []byte, v interface{}) error {
 	return yamlDecoder.NewYAMLToJSONDecoder(bytes.NewBuffer(data)).Decode(v)
 }
 
+// UnmarshalWithJSONDecoder takes a reader of yaml bytes converts the document or documents to JSON
```

Comment:
Could you tighten up the comment a bit? I am being so annoying because it's an exported function. 😺

```suggestion
// UnmarshalWithJSONDecoder expects a reader of raw YAML. 
// It converts the document or documents to JSON,
// then decodes the JSON bytes into a slice of values of type T.
// Type T must be a pointer, or the function will panic.
```

Reply:
Can do it's a good suggestion 👍🏾